### PR TITLE
Update deps & tests, remove old meta remnants

### PR DESCRIPTION
**Pull request by fluiderson**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,8 +1,69 @@
...
+use hashbrown::HashMap;
 
-// See `Cargo.toml` for the description of the "binary-vendor" feature.
-#[cfg(feature = "binary-vendor")]
+#[cfg(feature = "dummy")]
 include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+
+static mut STATE: Option<HashMap<ActorId, u128>> = None;
+
+fn static_mut_state() -> &'static mut HashMap<ActorId, u128> {
```
Comment (shamilsan) [inline src/lib.rs:18]: I'd rename it to `state_mut` or `get_state_mut`.

**Review by shamilsan — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@
 [![Build][build_badge]][build_href]
 [![License][lic_badge]][lic_href]
 
-[build_badge]: https://github.com/gear-dapps/app/workflows/Build/badge.svg
+[build_badge]: https://img.shields.io/github/actions/workflow/status/gear-dapps/app/build.yml?label=Build
```
[the remark this answers is not in the document]
Comment (fluiderson) [reply README.md:12]: Yes. IMO, the unified style looks better.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -22,7 +22,7 @@ jobs:
         uses: Swatinem/rust-cache@v2
 
       - name: Remove binaries from cache
-        run: rm -vf target/wasm32-unknown-unknown/release/*.wasm
+        run: rm -vfr target/wasm32-unknown-unknown/*
```
[the remark this answers is not in the document]
Comment (fluiderson) [reply .github/workflows/build.yml:25]: That's the `target/wasm32-unknown-unknown` directory, not `target`. So this change does the same - removes only WASM binaries without touching a build cache.

File: src/lib.rs [hunk 4]
```diff
@@ -1,8 +1,69 @@
...
+    match unsafe { &mut STATE } {
+        Some(state) => state,
+        None => unreachable!("state can't be uninitialized"),
+    }
```
[the remark this answers is not in the document]
Comment (fluiderson) [reply src/lib.rs:22]:
```rs
error[E0515]: cannot return value referencing temporary value
    --> src/lib.rs:20:5
     |
  20 |     unsafe { &mut STATE.expect("state can't be uninitialized") }
     |     ^^^^^^^^^^^^^^--------------------------------------------^^
     |     |             |
     |     |             temporary value created here
     |     returns a value referencing data owned by the current function
```

[the remark this answers is not in the document]
Comment (shamilsan) [reply src/lib.rs]: Ok,

```rust
unsafe { STATE.as_mut().expect("...") }
```

File: src/lib.rs [hunk 5]
```diff
@@ -1,8 +1,72 @@
...
 #[cfg(feature = "binary-vendor")]
 include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+
+static mut STATE: Option<HashMap<ActorId, u128>> = None;
+
+unsafe fn state_mut() -> &'static mut HashMap<ActorId, u128> {
+    match &mut STATE {
+        Some(state) => state,
+        None => unreachable_unchecked(),
+    }
```
Comment (shamilsan) [inline src/lib.rs:23]:
```suggestion
    unsafe { STATE.as_mut().expect("...") }
```

**Review by shamilsan — APPROVED** [review]

File: src/lib.rs [hunk 6]
```diff
@@ -1,8 +1,72 @@
...
+use hashbrown::HashMap;
+use hint::unreachable_unchecked;
 
-// See `Cargo.toml` for the description of the "binary-vendor" feature.
 #[cfg(feature = "binary-vendor")]
 include!(concat!(env!("OUT_DIR"), "/wasm_binary.rs"));
+
+static mut STATE: Option<HashMap<ActorId, u128>> = None;
+
+unsafe fn state_mut() -> &'static mut HashMap<ActorId, u128> {
```
Comment (shamilsan) [inline src/lib.rs:19]:
```suggestion
fn state_mut() -> &'static mut HashMap<ActorId, u128> {
```

File: src/lib.rs (hunk 5 above)
Comment (shamilsan) [reply src/lib.rs:23]: Or even [`unwrap_unchecked`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_unchecked) if we want compiler optimizes it.

**Review by azoyan — APPROVED** [review]

File: src/lib.rs (hunk 6 above)
Comment (fluiderson) [reply src/lib.rs:19]: I'd like to leave it unsafe since `.unchecked_unwrap()` can lead to UB if somebody will use this before a state initialization.

**Pull request merged**